// @aws-sdk/client-appsync@3.1041.0/dist-es/models/errors.js
import { AppSyncServiceException as __BaseException } from "./AppSyncServiceException";
export class AccessDeniedException extends __BaseException {
    name = "AccessDeniedException";
    $fault = "client";
    constructor(opts) {
        super({
            name: "AccessDeniedException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, AccessDeniedException.prototype);
    }
}
export class ApiKeyLimitExceededException extends __BaseException {
    name = "ApiKeyLimitExceededException";
    $fault = "client";
    constructor(opts) {
        super({
            name: "ApiKeyLimitExceededException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, ApiKeyLimitExceededException.prototype);
    }
}
export class ApiKeyValidityOutOfBoundsException extends __BaseException {
    name = "ApiKeyValidityOutOfBoundsException";
    $fault = "client";
    constructor(opts) {
        super({
            name: "ApiKeyValidityOutOfBoundsException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, ApiKeyValidityOutOfBoundsException.prototype);
    }
}
export class ApiLimitExceededException extends __BaseException {
    name = "ApiLimitExceededException";
    $fault = "client";
    constructor(opts) {
        super({
            name: "ApiLimitExceededException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, ApiLimitExceededException.prototype);
    }
}
export class BadRequestException extends __BaseException {
    name = "BadRequestException";
    $fault = "client";
    reason;
    detail;
    constructor(opts) {
        super({
            name: "BadRequestException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, BadRequestException.prototype);
        this.reason = opts.reason;
        this.detail = opts.detail;
    }
}
export class InternalFailureException extends __BaseException {
    name = "InternalFailureException";
    $fault = "server";
    constructor(opts) {
        super({
            name: "InternalFailureException",
            $fault: "server",
            ...opts,
        });
        Object.setPrototypeOf(this, InternalFailureException.prototype);
    }
}
export class NotFoundException extends __BaseException {
    name = "NotFoundException";
    $fault = "client";
    constructor(opts) {
        super({
            name: "NotFoundException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, NotFoundException.prototype);
    }
}
export class ConcurrentModificationException extends __BaseException {
    name = "ConcurrentModificationException";
    $fault = "client";
    constructor(opts) {
        super({
            name: "ConcurrentModificationException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, ConcurrentModificationException.prototype);
    }
}
export class LimitExceededException extends __BaseException {
    name = "LimitExceededException";
    $fault = "client";
    constructor(opts) {
        super({
            name: "LimitExceededException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, LimitExceededException.prototype);
    }
}
export class UnauthorizedException extends __BaseException {
    name = "UnauthorizedException";
    $fault = "client";
    constructor(opts) {
        super({
            name: "UnauthorizedException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, UnauthorizedException.prototype);
    }
}
export class ServiceQuotaExceededException extends __BaseException {
    name = "ServiceQuotaExceededException";
    $fault = "client";
    constructor(opts) {
        super({
            name: "ServiceQuotaExceededException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, ServiceQuotaExceededException.prototype);
    }
}
export class ConflictException extends __BaseException {
    name = "ConflictException";
    $fault = "client";
    constructor(opts) {
        super({
            name: "ConflictException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, ConflictException.prototype);
    }
}
export class GraphQLSchemaException extends __BaseException {
    name = "GraphQLSchemaException";
    $fault = "client";
    constructor(opts) {
        super({
            name: "GraphQLSchemaException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, GraphQLSchemaException.prototype);
    }
}
