// @aws-sdk/client-amplify@3.1041.0/dist-es/schemas/schemas_0.js
const _A = "App";
const _ABCC = "AutoBranchCreationConfig";
const _AT = "AccessToken";
const _Ap = "Apps";
const _Ar = "Artifact";
const _Art = "Artifacts";
const _B = "Backend";
const _BAC = "BasicAuthCredentials";
const _BE = "BackendEnvironment";
const _BEa = "BackendEnvironments";
const _BRE = "BadRequestException";
const _BS = "BuildSpec";
const _Br = "Branch";
const _Bra = "Branches";
const _C = "Certificate";
const _CA = "CreateApp";
const _CAR = "CreateAppRequest";
const _CARr = "CreateAppResult";
const _CB = "CreateBranch";
const _CBE = "CreateBackendEnvironment";
const _CBER = "CreateBackendEnvironmentRequest";
const _CBERr = "CreateBackendEnvironmentResult";
const _CBR = "CreateBranchRequest";
const _CBRr = "CreateBranchResult";
const _CC = "CacheConfig";
const _CD = "CreateDeployment";
const _CDA = "CreateDomainAssociation";
const _CDAR = "CreateDomainAssociationRequest";
const _CDARr = "CreateDomainAssociationResult";
const _CDR = "CreateDeploymentRequest";
const _CDRr = "CreateDeploymentResult";
const _CR = "CustomRule";
const _CRu = "CustomRules";
const _CS = "CertificateSettings";
const _CW = "CreateWebhook";
const _CWR = "CreateWebhookRequest";
const _CWRr = "CreateWebhookResult";
const _DA = "DomainAssociation";
const _DAR = "DeleteAppRequest";
const _DARe = "DeleteAppResult";
const _DAe = "DeleteApp";
const _DAo = "DomainAssociations";
const _DB = "DeleteBranch";
const _DBE = "DeleteBackendEnvironment";
const _DBER = "DeleteBackendEnvironmentRequest";
const _DBERe = "DeleteBackendEnvironmentResult";
const _DBR = "DeleteBranchRequest";
const _DBRe = "DeleteBranchResult";
const _DDA = "DeleteDomainAssociation";
const _DDAR = "DeleteDomainAssociationRequest";
const _DDARe = "DeleteDomainAssociationResult";
const _DJ = "DeleteJob";
const _DJR = "DeleteJobRequest";
const _DJRe = "DeleteJobResult";
const _DSFE = "DependentServiceFailureException";
const _DW = "DeleteWebhook";
const _DWR = "DeleteWebhookRequest";
const _DWRe = "DeleteWebhookResult";
const _GA = "GetApp";
const _GAL = "GenerateAccessLogs";
const _GALR = "GenerateAccessLogsRequest";
const _GALRe = "GenerateAccessLogsResult";
const _GAR = "GetAppRequest";
const _GARe = "GetAppResult";
const _GAU = "GetArtifactUrl";
const _GAUR = "GetArtifactUrlRequest";
const _GAURe = "GetArtifactUrlResult";
const _GB = "GetBranch";
const _GBE = "GetBackendEnvironment";
const _GBER = "GetBackendEnvironmentRequest";
const _GBERe = "GetBackendEnvironmentResult";
const _GBR = "GetBranchRequest";
const _GBRe = "GetBranchResult";
const _GDA = "GetDomainAssociation";
const _GDAR = "GetDomainAssociationRequest";
const _GDARe = "GetDomainAssociationResult";
const _GJ = "GetJob";
const _GJR = "GetJobRequest";
const _GJRe = "GetJobResult";
const _GW = "GetWebhook";
const _GWR = "GetWebhookRequest";
const _GWRe = "GetWebhookResult";
const _IFE = "InternalFailureException";
const _J = "Job";
const _JC = "JobConfig";
const _JS = "JobSummary";
const _JSo = "JobSummaries";
const _LA = "ListApps";
const _LAR = "ListAppsRequest";
const _LARi = "ListAppsResult";
const _LARis = "ListArtifactsRequest";
const _LARist = "ListArtifactsResult";
const _LAi = "ListArtifacts";
const _LB = "ListBranches";
const _LBE = "ListBackendEnvironments";
const _LBER = "ListBackendEnvironmentsRequest";
const _LBERi = "ListBackendEnvironmentsResult";
const _LBR = "ListBranchesRequest";
const _LBRi = "ListBranchesResult";
const _LDA = "ListDomainAssociations";
const _LDAR = "ListDomainAssociationsRequest";
const _LDARi = "ListDomainAssociationsResult";
const _LEE = "LimitExceededException";
const _LJ = "ListJobs";
const _LJR = "ListJobsRequest";
const _LJRi = "ListJobsResult";
const _LTFR = "ListTagsForResource";
const _LTFRR = "ListTagsForResourceRequest";
const _LTFRRi = "ListTagsForResourceResponse";
const _LW = "ListWebhooks";
const _LWR = "ListWebhooksRequest";
const _LWRi = "ListWebhooksResult";
const _NFE = "NotFoundException";
const _OT = "OauthToken";
const _PB = "ProductionBranch";
const _RNFE = "ResourceNotFoundException";
const _S = "Step";
const _SD = "SubDomain";
const _SDR = "StartDeploymentRequest";
const _SDRt = "StartDeploymentResult";
const _SDS = "SubDomainSetting";
const _SDSu = "SubDomainSettings";
const _SDt = "StartDeployment";
const _SDu = "SubDomains";
const _SJ = "StartJob";
const _SJR = "StartJobRequest";
const _SJRt = "StartJobResult";
const _SJRto = "StopJobRequest";
const _SJRtop = "StopJobResult";
const _SJt = "StopJob";
const _St = "Steps";
const _TR = "TagResource";
const _TRR = "TagResourceRequest";
const _TRRa = "TagResourceResponse";
const _UA = "UpdateApp";
const _UAR = "UpdateAppRequest";
const _UARp = "UpdateAppResult";
const _UB = "UpdateBranch";
const _UBR = "UpdateBranchRequest";
const _UBRp = "UpdateBranchResult";
const _UDA = "UpdateDomainAssociation";
const _UDAR = "UpdateDomainAssociationRequest";
const _UDARp = "UpdateDomainAssociationResult";
const _UE = "UnauthorizedException";
const _UR = "UntagResource";
const _URR = "UntagResourceRequest";
const _URRn = "UntagResourceResponse";
const _UW = "UpdateWebhook";
const _UWR = "UpdateWebhookRequest";
const _UWRp = "UpdateWebhookResult";
const _W = "Webhook";
const _WC = "WafConfiguration";
const _We = "Webhooks";
const _a = "app";
const _aA = "appArn";
const _aBCC = "autoBranchCreationConfig";
const _aBCP = "autoBranchCreationPatterns";
const _aFN = "artifactFileName";
const _aI = "appId";
const _aIr = "artifactId";
const _aJI = "activeJobId";
const _aR = "associatedResources";
const _aSDCP = "autoSubDomainCreationPatterns";
const _aSDIAMR = "autoSubDomainIAMRole";
const _aT = "accessToken";
const _aU = "artifactUrl";
const _aUr = "artifactsUrl";
const _ap = "apps";
const _ar = "artifacts";
const _b = "backend";
const _bA = "branchArn";
const _bAC = "basicAuthCredentials";
const _bCT = "buildComputeType";
const _bE = "backendEnvironment";
const _bEA = "backendEnvironmentArn";
const _bEa = "backendEnvironments";
const _bN = "branchName";
const _bS = "buildSpec";
const _br = "branch";
const _bra = "branches";
const _c = "client";
const _cC = "cacheConfig";
const _cCA = "customCertificateArn";
const _cD = "customDomains";
const _cH = "customHeaders";
const _cI = "commitId";
const _cM = "commitMessage";
const _cR = "customRules";
const _cRA = "computeRoleArn";
const _cS = "certificateSettings";
const _cT = "createTime";
const _cTo = "commitTime";
const _cVDNSR = "certificateVerificationDNSRecord";
const _ce = "certificate";
const _co = "code";
const _con = "condition";
const _cont = "context";
const _d = "description";
const _dA = "deploymentArtifacts";
const _dAA = "domainAssociationArn";
const _dAo = "domainAssociation";
const _dAom = "domainAssociations";
const _dB = "destinationBranch";
const _dD = "defaultDomain";
const _dN = "displayName";
const _dNo = "domainName";
const _dR = "dnsRecord";
const _dS = "domainStatus";
const _e = "error";
const _eAB = "enableAutoBuild";
const _eABC = "enableAutoBranchCreation";
const _eASD = "enableAutoSubDomain";
const _eBA = "enableBasicAuth";
const _eBAB = "enableBranchAutoBuild";
const _eBAD = "enableBranchAutoDeletion";
const _eN = "environmentName";
const _eNn = "enableNotification";
const _ePM = "enablePerformanceMode";
const _ePRP = "enablePullRequestPreview";
const _eSP = "enableSkewProtection";
const _eT = "endTime";
const _eV = "environmentVariables";
const _f = "framework";
const _fM = "fileMap";
const _fUU = "fileUploadUrls";
const _h = "http";
const _hE = "httpError";
const _hQ = "httpQuery";
const _iSRA = "iamServiceRoleArn";
const _j = "job";
const _jA = "jobArn";
const _jC = "jobConfig";
const _jI = "jobId";
const _jR = "jobReason";
const _jS = "jobSummary";
const _jSo = "jobSummaries";
const _jT = "jobType";
const _lDT = "lastDeployTime";
const _lU = "logUrl";
const _m = "message";
const _mR = "maxResults";
const _n = "name";
const _nT = "nextToken";
const _oT = "oauthToken";
const _p = "platform";
const _pB = "productionBranch";
const _pREN = "pullRequestEnvironmentName";
const _pr = "prefix";
const _r = "repository";
const _rA = "resourceArn";
const _rCM = "repositoryCloneMethod";
const _s = "smithy.ts.sdk.synthetic.com.amazonaws.amplify";
const _sA = "stackArn";
const _sB = "sourceBranch";
const _sD = "subDomains";
const _sDS = "subDomainSettings";
const _sDSu = "subDomainSetting";
const _sN = "stackName";
const _sNt = "stepName";
const _sR = "statusReason";
const _sT = "startTime";
const _sU = "sourceUrl";
const _sUT = "sourceUrlType";
const _sc = "screenshots";
const _se = "server";
const _so = "source";
const _st = "stage";
const _sta = "status";
const _ste = "steps";
const _su = "summary";
const _t = "tags";
const _tAU = "testArtifactsUrl";
const _tCU = "testConfigUrl";
const _tK = "tagKeys";
const _tNOJ = "totalNumberOfJobs";
const _tU = "thumbnailUrl";
const _ta = "target";
const _tt = "ttl";
const _ty = "type";
const _uS = "updateStatus";
const _uT = "updateTime";
const _v = "verified";
const _w = "webhook";
const _wA = "webhookArn";
const _wAA = "webAclArn";
const _wC = "wafConfiguration";
const _wCT = "webhookCreateTime";
const _wI = "webhookId";
const _wS = "wafStatus";
const _wU = "webhookUrl";
const _we = "webhooks";
const _zUU = "zipUploadUrl";
const n0 = "com.amazonaws.amplify";
import { TypeRegistry } from "@smithy/core/schema";
import { AmplifyServiceException } from "../models/AmplifyServiceException";
import { BadRequestException, DependentServiceFailureException, InternalFailureException, LimitExceededException, NotFoundException, ResourceNotFoundException, UnauthorizedException, } from "../models/errors";
const _s_registry = TypeRegistry.for(_s);
export var AmplifyServiceException$ = [-3, _s, "AmplifyServiceException", 0, [], []];
_s_registry.registerError(AmplifyServiceException$, AmplifyServiceException);
const n0_registry = TypeRegistry.for(n0);
export var BadRequestException$ = [-3, n0, _BRE,
    { [_e]: _c, [_hE]: 400 },
    [_m],
    [0]
];
n0_registry.registerError(BadRequestException$, BadRequestException);
export var DependentServiceFailureException$ = [-3, n0, _DSFE,
    { [_e]: _se, [_hE]: 503 },
    [_m],
    [0]
];
n0_registry.registerError(DependentServiceFailureException$, DependentServiceFailureException);
export var InternalFailureException$ = [-3, n0, _IFE,
    { [_e]: _se, [_hE]: 500 },
    [_m],
    [0]
];
n0_registry.registerError(InternalFailureException$, InternalFailureException);
export var LimitExceededException$ = [-3, n0, _LEE,
    { [_e]: _c, [_hE]: 429 },
    [_m],
    [0]
];
n0_registry.registerError(LimitExceededException$, LimitExceededException);
export var NotFoundException$ = [-3, n0, _NFE,
    { [_e]: _c, [_hE]: 404 },
    [_m],
    [0]
];
n0_registry.registerError(NotFoundException$, NotFoundException);
export var ResourceNotFoundException$ = [-3, n0, _RNFE,
    { [_e]: _c, [_hE]: 404 },
    [_co, _m],
    [0, 0], 2
];
n0_registry.registerError(ResourceNotFoundException$, ResourceNotFoundException);
export var UnauthorizedException$ = [-3, n0, _UE,
    { [_e]: _c, [_hE]: 401 },
    [_m],
    [0]
];
n0_registry.registerError(UnauthorizedException$, UnauthorizedException);
export const errorTypeRegistries = [
    _s_registry,
    n0_registry,
];
var AccessToken = [0, n0, _AT, 8, 0];
var BasicAuthCredentials = [0, n0, _BAC, 8, 0];
var BuildSpec = [0, n0, _BS, 8, 0];
var OauthToken = [0, n0, _OT, 8, 0];
export var App$ = [3, n0, _A,
    0,
    [_aI, _aA, _n, _d, _r, _p, _cT, _uT, _eV, _dD, _eBAB, _eBA, _t, _cRA, _iSRA, _eBAD, _bAC, _cR, _pB, _bS, _cH, _eABC, _aBCP, _aBCC, _rCM, _cC, _wCT, _wC, _jC],
    [0, 0, 0, 0, 0, 0, 4, 4, 128 | 0, 0, 2, 2, 128 | 0, 0, 0, 2, [() => BasicAuthCredentials, 0], () => CustomRules, () => ProductionBranch$, [() => BuildSpec, 0], 0, 2, 64 | 0, [() => AutoBranchCreationConfig$, 0], 0, () => CacheConfig$, 4, () => WafConfiguration$, () => JobConfig$], 12
];
export var Artifact$ = [3, n0, _Ar,
    0,
    [_aFN, _aIr],
    [0, 0], 2
];
export var AutoBranchCreationConfig$ = [3, n0, _ABCC,
    0,
    [_st, _f, _eAB, _eV, _bAC, _eBA, _ePM, _bS, _ePRP, _pREN],
    [0, 0, 2, 128 | 0, [() => BasicAuthCredentials, 0], 2, 2, [() => BuildSpec, 0], 2, 0]
];
export var Backend$ = [3, n0, _B,
    0,
    [_sA],
    [0]
];
export var BackendEnvironment$ = [3, n0, _BE,
    0,
    [_bEA, _eN, _cT, _uT, _sN, _dA],
    [0, 0, 4, 4, 0, 0], 4
];
export var Branch$ = [3, n0, _Br,
    0,
    [_bA, _bN, _d, _st, _dN, _eNn, _cT, _uT, _eV, _eAB, _cD, _f, _aJI, _tNOJ, _eBA, _tt, _ePRP, _t, _eSP, _ePM, _tU, _bAC, _bS, _aR, _pREN, _dB, _sB, _bEA, _b, _cRA],
    [0, 0, 0, 0, 0, 2, 4, 4, 128 | 0, 2, 64 | 0, 0, 0, 0, 2, 0, 2, 128 | 0, 2, 2, 0, [() => BasicAuthCredentials, 0], [() => BuildSpec, 0], 64 | 0, 0, 0, 0, 0, () => Backend$, 0], 17
];
export var CacheConfig$ = [3, n0, _CC,
    0,
    [_ty],
    [0], 1
];
export var Certificate$ = [3, n0, _C,
    0,
    [_ty, _cCA, _cVDNSR],
    [0, 0, 0], 1
];
export var CertificateSettings$ = [3, n0, _CS,
    0,
    [_ty, _cCA],
    [0, 0], 1
];
export var CreateAppRequest$ = [3, n0, _CAR,
    0,
    [_n, _d, _r, _p, _cRA, _iSRA, _oT, _aT, _eV, _eBAB, _eBAD, _eBA, _bAC, _cR, _t, _bS, _cH, _eABC, _aBCP, _aBCC, _jC, _cC],
    [0, 0, 0, 0, 0, 0, [() => OauthToken, 0], [() => AccessToken, 0], 128 | 0, 2, 2, 2, [() => BasicAuthCredentials, 0], () => CustomRules, 128 | 0, [() => BuildSpec, 0], 0, 2, 64 | 0, [() => AutoBranchCreationConfig$, 0], () => JobConfig$, () => CacheConfig$], 1
];
export var CreateAppResult$ = [3, n0, _CARr,
    0,
    [_a],
    [[() => App$, 0]], 1
];
export var CreateBackendEnvironmentRequest$ = [3, n0, _CBER,
    0,
    [_aI, _eN, _sN, _dA],
    [[0, 1], 0, 0, 0], 2
];
export var CreateBackendEnvironmentResult$ = [3, n0, _CBERr,
    0,
    [_bE],
    [() => BackendEnvironment$], 1
];
export var CreateBranchRequest$ = [3, n0, _CBR,
    0,
    [_aI, _bN, _d, _st, _f, _eNn, _eAB, _eSP, _eV, _bAC, _eBA, _ePM, _t, _bS, _tt, _dN, _ePRP, _pREN, _bEA, _b, _cRA],
    [[0, 1], 0, 0, 0, 0, 2, 2, 2, 128 | 0, [() => BasicAuthCredentials, 0], 2, 2, 128 | 0, [() => BuildSpec, 0], 0, 0, 2, 0, 0, () => Backend$, 0], 2
];
export var CreateBranchResult$ = [3, n0, _CBRr,
    0,
    [_br],
    [[() => Branch$, 0]], 1
];
export var CreateDeploymentRequest$ = [3, n0, _CDR,
    0,
    [_aI, _bN, _fM],
    [[0, 1], [0, 1], 128 | 0], 2
];
export var CreateDeploymentResult$ = [3, n0, _CDRr,
    0,
    [_fUU, _zUU, _jI],
    [128 | 0, 0, 0], 2
];
export var CreateDomainAssociationRequest$ = [3, n0, _CDAR,
    0,
    [_aI, _dNo, _sDS, _eASD, _aSDCP, _aSDIAMR, _cS],
    [[0, 1], 0, () => SubDomainSettings, 2, 64 | 0, 0, () => CertificateSettings$], 3
];
export var CreateDomainAssociationResult$ = [3, n0, _CDARr,
    0,
    [_dAo],
    [() => DomainAssociation$], 1
];
export var CreateWebhookRequest$ = [3, n0, _CWR,
    0,
    [_aI, _bN, _d],
    [[0, 1], 0, 0], 2
];
export var CreateWebhookResult$ = [3, n0, _CWRr,
    0,
    [_w],
    [() => Webhook$], 1
];
export var CustomRule$ = [3, n0, _CR,
    0,
    [_so, _ta, _sta, _con],
    [0, 0, 0, 0], 2
];
export var DeleteAppRequest$ = [3, n0, _DAR,
    0,
    [_aI],
    [[0, 1]], 1
];
export var DeleteAppResult$ = [3, n0, _DARe,
    0,
    [_a],
    [[() => App$, 0]], 1
];
export var DeleteBackendEnvironmentRequest$ = [3, n0, _DBER,
    0,
    [_aI, _eN],
    [[0, 1], [0, 1]], 2
];
export var DeleteBackendEnvironmentResult$ = [3, n0, _DBERe,
    0,
    [_bE],
    [() => BackendEnvironment$], 1
];
export var DeleteBranchRequest$ = [3, n0, _DBR,
    0,
    [_aI, _bN],
    [[0, 1], [0, 1]], 2
];
export var DeleteBranchResult$ = [3, n0, _DBRe,
    0,
    [_br],
    [[() => Branch$, 0]], 1
];
export var DeleteDomainAssociationRequest$ = [3, n0, _DDAR,
    0,
    [_aI, _dNo],
    [[0, 1], [0, 1]], 2
];
export var DeleteDomainAssociationResult$ = [3, n0, _DDARe,
    0,
    [_dAo],
    [() => DomainAssociation$], 1
];
export var DeleteJobRequest$ = [3, n0, _DJR,
    0,
    [_aI, _bN, _jI],
    [[0, 1], [0, 1], [0, 1]], 3
];
export var DeleteJobResult$ = [3, n0, _DJRe,
    0,
    [_jS],
    [() => JobSummary$], 1
];
export var DeleteWebhookRequest$ = [3, n0, _DWR,
    0,
    [_wI],
    [[0, 1]], 1
];
export var DeleteWebhookResult$ = [3, n0, _DWRe,
    0,
    [_w],
    [() => Webhook$], 1
];
export var DomainAssociation$ = [3, n0, _DA,
    0,
    [_dAA, _dNo, _eASD, _dS, _sR, _sD, _aSDCP, _aSDIAMR, _uS, _cVDNSR, _ce],
    [0, 0, 2, 0, 0, () => SubDomains, 64 | 0, 0, 0, 0, () => Certificate$], 6
];
export var GenerateAccessLogsRequest$ = [3, n0, _GALR,
    0,
    [_dNo, _aI, _sT, _eT],
    [0, [0, 1], 4, 4], 2
];
export var GenerateAccessLogsResult$ = [3, n0, _GALRe,
    0,
    [_lU],
    [0]
];
export var GetAppRequest$ = [3, n0, _GAR,
    0,
    [_aI],
    [[0, 1]], 1
];
export var GetAppResult$ = [3, n0, _GARe,
    0,
    [_a],
    [[() => App$, 0]], 1
];
export var GetArtifactUrlRequest$ = [3, n0, _GAUR,
    0,
    [_aIr],
    [[0, 1]], 1
];
export var GetArtifactUrlResult$ = [3, n0, _GAURe,
    0,
    [_aIr, _aU],
    [0, 0], 2
];
export var GetBackendEnvironmentRequest$ = [3, n0, _GBER,
    0,
    [_aI, _eN],
    [[0, 1], [0, 1]], 2
];
export var GetBackendEnvironmentResult$ = [3, n0, _GBERe,
    0,
    [_bE],
    [() => BackendEnvironment$], 1
];
export var GetBranchRequest$ = [3, n0, _GBR,
    0,
    [_aI, _bN],
    [[0, 1], [0, 1]], 2
];
export var GetBranchResult$ = [3, n0, _GBRe,
    0,
    [_br],
    [[() => Branch$, 0]], 1
];
export var GetDomainAssociationRequest$ = [3, n0, _GDAR,
    0,
    [_aI, _dNo],
    [[0, 1], [0, 1]], 2
];
export var GetDomainAssociationResult$ = [3, n0, _GDARe,
    0,
    [_dAo],
    [() => DomainAssociation$], 1
];
export var GetJobRequest$ = [3, n0, _GJR,
    0,
    [_aI, _bN, _jI],
    [[0, 1], [0, 1], [0, 1]], 3
];
export var GetJobResult$ = [3, n0, _GJRe,
    0,
    [_j],
    [() => Job$], 1
];
export var GetWebhookRequest$ = [3, n0, _GWR,
    0,
    [_wI],
    [[0, 1]], 1
];
export var GetWebhookResult$ = [3, n0, _GWRe,
    0,
    [_w],
    [() => Webhook$], 1
];
export var Job$ = [3, n0, _J,
    0,
    [_su, _ste],
    [() => JobSummary$, () => Steps], 2
];
export var JobConfig$ = [3, n0, _JC,
    0,
    [_bCT],
    [0], 1
];
export var JobSummary$ = [3, n0, _JS,
    0,
    [_jA, _jI, _cI, _cM, _cTo, _sT, _sta, _jT, _eT, _sU, _sUT],
    [0, 0, 0, 0, 4, 4, 0, 0, 4, 0, 0], 8
];
export var ListAppsRequest$ = [3, n0, _LAR,
    0,
    [_nT, _mR],
    [[0, { [_hQ]: _nT }], [1, { [_hQ]: _mR }]]
];
export var ListAppsResult$ = [3, n0, _LARi,
    0,
    [_ap, _nT],
    [[() => Apps, 0], 0], 1
];
export var ListArtifactsRequest$ = [3, n0, _LARis,
    0,
    [_aI, _bN, _jI, _nT, _mR],
    [[0, 1], [0, 1], [0, 1], [0, { [_hQ]: _nT }], [1, { [_hQ]: _mR }]], 3
];
export var ListArtifactsResult$ = [3, n0, _LARist,
    0,
    [_ar, _nT],
    [() => Artifacts, 0], 1
];
export var ListBackendEnvironmentsRequest$ = [3, n0, _LBER,
    0,
    [_aI, _eN, _nT, _mR],
    [[0, 1], [0, { [_hQ]: _eN }], [0, { [_hQ]: _nT }], [1, { [_hQ]: _mR }]], 1
];
export var ListBackendEnvironmentsResult$ = [3, n0, _LBERi,
    0,
    [_bEa, _nT],
    [() => BackendEnvironments, 0], 1
];
export var ListBranchesRequest$ = [3, n0, _LBR,
    0,
    [_aI, _nT, _mR],
    [[0, 1], [0, { [_hQ]: _nT }], [1, { [_hQ]: _mR }]], 1
];
export var ListBranchesResult$ = [3, n0, _LBRi,
    0,
    [_bra, _nT],
    [[() => Branches, 0], 0], 1
];
export var ListDomainAssociationsRequest$ = [3, n0, _LDAR,
    0,
    [_aI, _nT, _mR],
    [[0, 1], [0, { [_hQ]: _nT }], [1, { [_hQ]: _mR }]], 1
];
export var ListDomainAssociationsResult$ = [3, n0, _LDARi,
    0,
    [_dAom, _nT],
    [() => DomainAssociations, 0], 1
];
export var ListJobsRequest$ = [3, n0, _LJR,
    0,
    [_aI, _bN, _nT, _mR],
    [[0, 1], [0, 1], [0, { [_hQ]: _nT }], [1, { [_hQ]: _mR }]], 2
];
export var ListJobsResult$ = [3, n0, _LJRi,
    0,
    [_jSo, _nT],
    [() => JobSummaries, 0], 1
];
export var ListTagsForResourceRequest$ = [3, n0, _LTFRR,
    0,
    [_rA],
    [[0, 1]], 1
];
export var ListTagsForResourceResponse$ = [3, n0, _LTFRRi,
    0,
    [_t],
    [128 | 0]
];
export var ListWebhooksRequest$ = [3, n0, _LWR,
    0,
    [_aI, _nT, _mR],
    [[0, 1], [0, { [_hQ]: _nT }], [1, { [_hQ]: _mR }]], 1
];
export var ListWebhooksResult$ = [3, n0, _LWRi,
    0,
    [_we, _nT],
    [() => Webhooks, 0], 1
];
export var ProductionBranch$ = [3, n0, _PB,
    0,
    [_lDT, _sta, _tU, _bN],
    [4, 0, 0, 0]
];
export var StartDeploymentRequest$ = [3, n0, _SDR,
    0,
    [_aI, _bN, _jI, _sU, _sUT],
    [[0, 1], [0, 1], 0, 0, 0], 2
];
export var StartDeploymentResult$ = [3, n0, _SDRt,
    0,
    [_jS],
    [() => JobSummary$], 1
];
export var StartJobRequest$ = [3, n0, _SJR,
    0,
    [_aI, _bN, _jT, _jI, _jR, _cI, _cM, _cTo],
    [[0, 1], [0, 1], 0, 0, 0, 0, 0, 4], 3
];
export var StartJobResult$ = [3, n0, _SJRt,
    0,
    [_jS],
    [() => JobSummary$], 1
];
export var Step$ = [3, n0, _S,
    0,
    [_sNt, _sT, _sta, _eT, _lU, _aUr, _tAU, _tCU, _sc, _sR, _cont],
    [0, 4, 0, 4, 0, 0, 0, 0, 128 | 0, 0, 0], 4
];
export var StopJobRequest$ = [3, n0, _SJRto,
    0,
    [_aI, _bN, _jI],
    [[0, 1], [0, 1], [0, 1]], 3
];
export var StopJobResult$ = [3, n0, _SJRtop,
    0,
    [_jS],
    [() => JobSummary$], 1
];
export var SubDomain$ = [3, n0, _SD,
    0,
    [_sDSu, _v, _dR],
    [() => SubDomainSetting$, 2, 0], 3
];
export var SubDomainSetting$ = [3, n0, _SDS,
    0,
    [_pr, _bN],
    [0, 0], 2
];
export var TagResourceRequest$ = [3, n0, _TRR,
    0,
    [_rA, _t],
    [[0, 1], 128 | 0], 2
];
export var TagResourceResponse$ = [3, n0, _TRRa,
    0,
    [],
    []
];
export var UntagResourceRequest$ = [3, n0, _URR,
    0,
    [_rA, _tK],
    [[0, 1], [64 | 0, { [_hQ]: _tK }]], 2
];
export var UntagResourceResponse$ = [3, n0, _URRn,
    0,
    [],
    []
];
export var UpdateAppRequest$ = [3, n0, _UAR,
    0,
    [_aI, _n, _d, _p, _cRA, _iSRA, _eV, _eBAB, _eBAD, _eBA, _bAC, _cR, _bS, _cH, _eABC, _aBCP, _aBCC, _r, _oT, _aT, _jC, _cC],
    [[0, 1], 0, 0, 0, 0, 0, 128 | 0, 2, 2, 2, [() => BasicAuthCredentials, 0], () => CustomRules, [() => BuildSpec, 0], 0, 2, 64 | 0, [() => AutoBranchCreationConfig$, 0], 0, [() => OauthToken, 0], [() => AccessToken, 0], () => JobConfig$, () => CacheConfig$], 1
];
export var UpdateAppResult$ = [3, n0, _UARp,
    0,
    [_a],
    [[() => App$, 0]], 1
];
export var UpdateBranchRequest$ = [3, n0, _UBR,
    0,
    [_aI, _bN, _d, _f, _st, _eNn, _eAB, _eSP, _eV, _bAC, _eBA, _ePM, _bS, _tt, _dN, _ePRP, _pREN, _bEA, _b, _cRA],
    [[0, 1], [0, 1], 0, 0, 0, 2, 2, 2, 128 | 0, [() => BasicAuthCredentials, 0], 2, 2, [() => BuildSpec, 0], 0, 0, 2, 0, 0, () => Backend$, 0], 2
];
export var UpdateBranchResult$ = [3, n0, _UBRp,
    0,
    [_br],
    [[() => Branch$, 0]], 1
];
export var UpdateDomainAssociationRequest$ = [3, n0, _UDAR,
    0,
    [_aI, _dNo, _eASD, _sDS, _aSDCP, _aSDIAMR, _cS],
    [[0, 1], [0, 1], 2, () => SubDomainSettings, 64 | 0, 0, () => CertificateSettings$], 2
];
export var UpdateDomainAssociationResult$ = [3, n0, _UDARp,
    0,
    [_dAo],
    [() => DomainAssociation$], 1
];
export var UpdateWebhookRequest$ = [3, n0, _UWR,
    0,
    [_wI, _bN, _d],
    [[0, 1], 0, 0], 1
];
export var UpdateWebhookResult$ = [3, n0, _UWRp,
    0,
    [_w],
    [() => Webhook$], 1
];
export var WafConfiguration$ = [3, n0, _WC,
    0,
    [_wAA, _wS, _sR],
    [0, 0, 0]
];
export var Webhook$ = [3, n0, _W,
    0,
    [_wA, _wI, _wU, _bN, _d, _cT, _uT, _aI],
    [0, 0, 0, 0, 0, 4, 4, 0], 7
];
var Apps = [1, n0, _Ap,
    0, [() => App$,
        0]
];
var Artifacts = [1, n0, _Art,
    0, () => Artifact$
];
var AssociatedResources = 64 | 0;
var AutoBranchCreationPatterns = 64 | 0;
var AutoSubDomainCreationPatterns = 64 | 0;
var BackendEnvironments = [1, n0, _BEa,
    0, () => BackendEnvironment$
];
var Branches = [1, n0, _Bra,
    0, [() => Branch$,
        0]
];
var CustomDomains = 64 | 0;
var CustomRules = [1, n0, _CRu,
    0, () => CustomRule$
];
var DomainAssociations = [1, n0, _DAo,
    0, () => DomainAssociation$
];
var JobSummaries = [1, n0, _JSo,
    0, () => JobSummary$
];
var Steps = [1, n0, _St,
    0, () => Step$
];
var SubDomains = [1, n0, _SDu,
    0, () => SubDomain$
];
var SubDomainSettings = [1, n0, _SDSu,
    0, () => SubDomainSetting$
];
var TagKeyList = 64 | 0;
var Webhooks = [1, n0, _We,
    0, () => Webhook$
];
var EnvironmentVariables = 128 | 0;
var FileMap = 128 | 0;
var FileUploadUrls = 128 | 0;
var Screenshots = 128 | 0;
var TagMap = 128 | 0;
export var CreateApp$ = [9, n0, _CA,
    { [_h]: ["POST", "/apps", 200] }, () => CreateAppRequest$, () => CreateAppResult$
];
export var CreateBackendEnvironment$ = [9, n0, _CBE,
    { [_h]: ["POST", "/apps/{appId}/backendenvironments", 200] }, () => CreateBackendEnvironmentRequest$, () => CreateBackendEnvironmentResult$
];
export var CreateBranch$ = [9, n0, _CB,
    { [_h]: ["POST", "/apps/{appId}/branches", 200] }, () => CreateBranchRequest$, () => CreateBranchResult$
];
export var CreateDeployment$ = [9, n0, _CD,
    { [_h]: ["POST", "/apps/{appId}/branches/{branchName}/deployments", 200] }, () => CreateDeploymentRequest$, () => CreateDeploymentResult$
];
export var CreateDomainAssociation$ = [9, n0, _CDA,
    { [_h]: ["POST", "/apps/{appId}/domains", 200] }, () => CreateDomainAssociationRequest$, () => CreateDomainAssociationResult$
];
export var CreateWebhook$ = [9, n0, _CW,
    { [_h]: ["POST", "/apps/{appId}/webhooks", 200] }, () => CreateWebhookRequest$, () => CreateWebhookResult$
];
export var DeleteApp$ = [9, n0, _DAe,
    { [_h]: ["DELETE", "/apps/{appId}", 200] }, () => DeleteAppRequest$, () => DeleteAppResult$
];
export var DeleteBackendEnvironment$ = [9, n0, _DBE,
    { [_h]: ["DELETE", "/apps/{appId}/backendenvironments/{environmentName}", 200] }, () => DeleteBackendEnvironmentRequest$, () => DeleteBackendEnvironmentResult$
];
export var DeleteBranch$ = [9, n0, _DB,
    { [_h]: ["DELETE", "/apps/{appId}/branches/{branchName}", 200] }, () => DeleteBranchRequest$, () => DeleteBranchResult$
];
export var DeleteDomainAssociation$ = [9, n0, _DDA,
    { [_h]: ["DELETE", "/apps/{appId}/domains/{domainName}", 200] }, () => DeleteDomainAssociationRequest$, () => DeleteDomainAssociationResult$
];
export var DeleteJob$ = [9, n0, _DJ,
    { [_h]: ["DELETE", "/apps/{appId}/branches/{branchName}/jobs/{jobId}", 200] }, () => DeleteJobRequest$, () => DeleteJobResult$
];
export var DeleteWebhook$ = [9, n0, _DW,
    { [_h]: ["DELETE", "/webhooks/{webhookId}", 200] }, () => DeleteWebhookRequest$, () => DeleteWebhookResult$
];
export var GenerateAccessLogs$ = [9, n0, _GAL,
    { [_h]: ["POST", "/apps/{appId}/accesslogs", 200] }, () => GenerateAccessLogsRequest$, () => GenerateAccessLogsResult$
];
export var GetApp$ = [9, n0, _GA,
    { [_h]: ["GET", "/apps/{appId}", 200] }, () => GetAppRequest$, () => GetAppResult$
];
export var GetArtifactUrl$ = [9, n0, _GAU,
    { [_h]: ["GET", "/artifacts/{artifactId}", 200] }, () => GetArtifactUrlRequest$, () => GetArtifactUrlResult$
];
export var GetBackendEnvironment$ = [9, n0, _GBE,
    { [_h]: ["GET", "/apps/{appId}/backendenvironments/{environmentName}", 200] }, () => GetBackendEnvironmentRequest$, () => GetBackendEnvironmentResult$
];
export var GetBranch$ = [9, n0, _GB,
    { [_h]: ["GET", "/apps/{appId}/branches/{branchName}", 200] }, () => GetBranchRequest$, () => GetBranchResult$
];
export var GetDomainAssociation$ = [9, n0, _GDA,
    { [_h]: ["GET", "/apps/{appId}/domains/{domainName}", 200] }, () => GetDomainAssociationRequest$, () => GetDomainAssociationResult$
];
export var GetJob$ = [9, n0, _GJ,
    { [_h]: ["GET", "/apps/{appId}/branches/{branchName}/jobs/{jobId}", 200] }, () => GetJobRequest$, () => GetJobResult$
];
export var GetWebhook$ = [9, n0, _GW,
    { [_h]: ["GET", "/webhooks/{webhookId}", 200] }, () => GetWebhookRequest$, () => GetWebhookResult$
];
export var ListApps$ = [9, n0, _LA,
    { [_h]: ["GET", "/apps", 200] }, () => ListAppsRequest$, () => ListAppsResult$
];
export var ListArtifacts$ = [9, n0, _LAi,
    { [_h]: ["GET", "/apps/{appId}/branches/{branchName}/jobs/{jobId}/artifacts", 200] }, () => ListArtifactsRequest$, () => ListArtifactsResult$
];
export var ListBackendEnvironments$ = [9, n0, _LBE,
    { [_h]: ["GET", "/apps/{appId}/backendenvironments", 200] }, () => ListBackendEnvironmentsRequest$, () => ListBackendEnvironmentsResult$
];
export var ListBranches$ = [9, n0, _LB,
    { [_h]: ["GET", "/apps/{appId}/branches", 200] }, () => ListBranchesRequest$, () => ListBranchesResult$
];
export var ListDomainAssociations$ = [9, n0, _LDA,
    { [_h]: ["GET", "/apps/{appId}/domains", 200] }, () => ListDomainAssociationsRequest$, () => ListDomainAssociationsResult$
];
export var ListJobs$ = [9, n0, _LJ,
    { [_h]: ["GET", "/apps/{appId}/branches/{branchName}/jobs", 200] }, () => ListJobsRequest$, () => ListJobsResult$
];
export var ListTagsForResource$ = [9, n0, _LTFR,
    { [_h]: ["GET", "/tags/{resourceArn}", 200] }, () => ListTagsForResourceRequest$, () => ListTagsForResourceResponse$
];
export var ListWebhooks$ = [9, n0, _LW,
    { [_h]: ["GET", "/apps/{appId}/webhooks", 200] }, () => ListWebhooksRequest$, () => ListWebhooksResult$
];
export var StartDeployment$ = [9, n0, _SDt,
    { [_h]: ["POST", "/apps/{appId}/branches/{branchName}/deployments/start", 200] }, () => StartDeploymentRequest$, () => StartDeploymentResult$
];
export var StartJob$ = [9, n0, _SJ,
    { [_h]: ["POST", "/apps/{appId}/branches/{branchName}/jobs", 200] }, () => StartJobRequest$, () => StartJobResult$
];
export var StopJob$ = [9, n0, _SJt,
    { [_h]: ["DELETE", "/apps/{appId}/branches/{branchName}/jobs/{jobId}/stop", 200] }, () => StopJobRequest$, () => StopJobResult$
];
export var TagResource$ = [9, n0, _TR,
    { [_h]: ["POST", "/tags/{resourceArn}", 200] }, () => TagResourceRequest$, () => TagResourceResponse$
];
export var UntagResource$ = [9, n0, _UR,
    { [_h]: ["DELETE", "/tags/{resourceArn}", 200] }, () => UntagResourceRequest$, () => UntagResourceResponse$
];
export var UpdateApp$ = [9, n0, _UA,
    { [_h]: ["POST", "/apps/{appId}", 200] }, () => UpdateAppRequest$, () => UpdateAppResult$
];
export var UpdateBranch$ = [9, n0, _UB,
    { [_h]: ["POST", "/apps/{appId}/branches/{branchName}", 200] }, () => UpdateBranchRequest$, () => UpdateBranchResult$
];
export var UpdateDomainAssociation$ = [9, n0, _UDA,
    { [_h]: ["POST", "/apps/{appId}/domains/{domainName}", 200] }, () => UpdateDomainAssociationRequest$, () => UpdateDomainAssociationResult$
];
export var UpdateWebhook$ = [9, n0, _UW,
    { [_h]: ["POST", "/webhooks/{webhookId}", 200] }, () => UpdateWebhookRequest$, () => UpdateWebhookResult$
];
